### PR TITLE
doc: remove all hard coded version numbers to simplify release

### PR DESCRIPTION
**Pull request by monperrus**
baby pr for #2779

### CODE REVIEW

File: doc/_config.yml [hunk 1]
```diff
@@ -18,9 +18,16 @@ include: [_includes]
 # variables
 
 sidebar_tagline: Spoon
-sidebar_version: version 7.0.0
```
Comment (surli) [inline doc/_config.yml]: just for info, can't you define sidebar_version according to spoon_release?

File: doc/_config.yml [hunk 2]
```diff
@@ -18,9 +18,16 @@ include: [_includes]
...
-spoon_release: "7.1.0"
-spoon_snapshot: "7.2.0-SNAPSHOT"
+
+# this is replaced at deployment time by job https://ci.inria.fr/sos/job/Website Deployer/
+# LATESTVERSION=`curl -s "http://search.maven.org/solrsearch/select?q=g:%22fr.inria.gforge.spoon%22+AND+a:%22spoon-core%22" | jq .response.docs[0].latestVersion`
+# sed -i -e "s/^spoon_release: .*/spoon_release: $LATESTVERSION/" _config.yml
```
Comment (surli) [inline doc/_config.yml]: Is this command pushed somewhere?

**Comment by tdurieux** [comment]
@GerardPaligot did before a script that automatically updates those version. I feel it is a better solution

File: doc/_config.yml (hunk 1 above)
Comment (monperrus) [reply doc/_config.yml]: that is another option, but I prefer to simply get rid of sidebar_version

File: doc/_config.yml (hunk 2 above)
Comment (monperrus) [reply doc/_config.yml]: yes, in the Jenkins config, which is mirrored here https://github.com/SpoonLabs/spoon-ci-config

**Comment by monperrus** [comment]
You're right, the problem is that this script is a bash monster that hasn't been executed and maintained for more than 4 years.

**Comment by tdurieux** [comment]
I just looked a the script (that is not really a monster), it seems that it is pretty simple to maintain and only minor changes are required.
see: 
https://github.com/INRIA/spoon/blob/master/doc/_release/deploy_production.sh

**Comment by monperrus** [comment]
Thanks for the link.

Regardless of the script, I'm in favor of the simplification of this PR, on the ground that simpler means less error-prone today and in the future.

File: doc/_config.yml (hunk 2 above)
Comment (surli) [reply doc/_config.yml]: Good to know!

File: doc/_config.yml (hunk 1 above)
Comment (surli) [reply doc/_config.yml]: ok

**Comment by GerardPaligot** [comment]
That could be interesting to migrate all these old scripts from bash to a better and more maintainable language. You always use Maven to package your project, that could be interesting to migration to Gradle and create better scripts in Groovy or Kotlin for all these tasks that could be automatic. :)

**Comment by monperrus** [comment]
+100 for @GerardPaligot's comment. (good to see you Gérard :-)

For the next release, I'm considering writing some Python scripts.

**Comment by monperrus** [comment]
With no other comments, ready for merge.

**Comment by monperrus** [comment]
ready for merge

**Pull request merged**